web/generate-api-typescript-client-pre-commit.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -eo
+
+PATTERN="api/src/features/*"
+PATTERN+="|api/src/entities/*"
+
+CHANGED_API_FILES=()
+while read; do
+  CHANGED_API_FILES+=("$REPLY")
+done < <(echo "$(git diff --name-only --cached --diff-filter=ACMR)" | grep -E $PATTERN)
+
+if [ ${#CHANGED_API_FILES[@]} -eq 0 ]; then
+    echo "No changes in API relevant files. No need to generate API."
+else
+    echo "Changes detected in API relevant files. Generating API ..."
+    # This requires the API to be running on localhost port 5000
+    cd web
+    yarn openapi -i http://localhost:5000/openapi.json -o "$PWD/src/api/generated"
```

Comment:
How will this be handled in the CI? Or what error message is provided if localhost is not running?

Reply:
Since the CI will not see any "CHANGED_API_FILES", it won't run in the CI anyways. 
So having this in the CI doesn't actually solve anything, but that's "ok" for now. I guess we could add some sort of "--all-files" flag somehow, but then we'd need to start storing openapi.json as a file, which would lead to additional issues.